### PR TITLE
#708 account creation issues

### PR DESCRIPTION
**Pull request by tomlovesgithub**
It was previously possible for an account to be created without the user being progressed from the 'sign up' page. 
This was causing an error which stated that there already existed an account with the given email address.
In investigating I encountered some issues in the code i believe could have been leading to this behaviour. 
These issues surrounded the error monitoring on the form as well as password validation and a `isValid()` function which was not being called asynchronously, and i believe was submitting the form but not reloading the page.

---

- Add `await` into account creation, to prevent account being created without page being reloaded
- improve password validation and run validation handling on input for passwords
- Add `.catch` statement to prevent errors going uncaught
- Sort chronology so sign up actions are the last thing to happen before page reloads.

---
## Who should test?
🟩 Product owner
🟩 Developers
🟩 UTG
✅ All of the above

## How to test?
- Log out
- Go to the sign-in page.
- using a new (doesn't have to be real) email address begin the sign up process 
[please use the same email address throughout testing until you are given confirmation that you have successfully signed up]
- [ ] Note that the password box now updates with validation suggestions as the password is entered
- Attempt to sign up with a bad password 
- Attempt to sign up with a missing field in your DOB
- Attempt to sign up with any missing field

- Attempt to sign up immediately after satisfying any issues with the form

🟠 Medium risk -  while this doesn't change code that is shared with a lot of areas, its the page where users create an account which grants them access, we dont want this to go wrong!

### CODE REVIEW

**Comment by warrensearle** [comment]
Hey @tomlovesgithub the unit tests need fixing 😄

**Review by warrensearle — COMMENTED** [review]
Hey @tomlovesgithub the unit tests need fixing 😄

**Review by HalcyonJAC — APPROVED** [review]
Hi @tomlovesgithub 

I tested it locally and I can sign-up fine still. Unit tests all pass too. My only reservation is that the `signUp()` function involves several db ops which I think would be better if they were wrapped in a db transaction, so that in the event that something does wrong with one of them the user can at least attempt to sign-up again. 

In this instance though it would be difficult to implement I think since the first 2 ops (`createUserWithEmailAndPassword()` and `this.$store.dispatch('auth/setCurrentUser', userCredential.user)`) involve the auth database, whereas the next 2 ops (`this.$store.dispatch('candidate/create'...` and `this.$store.dispatch('candidate/savePersonalDetails'...`) involve the Firestore database and I don't think it is possible to create a database transactions across 2 databases.

So...I'm approving this now... ;)
`
Darrel.

**Review by mbrookeswebdev — APPROVED** [review]
Works well!

File: src/views/SignUp.vue [hunk 1]
```diff
@@ -182,6 +183,10 @@ export default {
         dateOfBirth: this.formData.dateOfBirth,
         nationalInsuranceNumber: this.formData.nationalInsuranceNumber,
       };
+      await this.$store.dispatch('auth/setCurrentUser', userCredential.user);
```
[the remark this answers is not in the document]
Comment (tomlovesgithub) [reply src/views/SignUp.vue:186]: I really agree with this, as the complex chronology of things here is what i believe is the root of the issue, batching would be ideal.
My only reservation might be to leave that to another ticket? 
Open to your suggestions if you think its worth adding in here,

**Review by KateMJAC — COMMENTED** [review]
This worked as it should with validation and correct behavour on sign-up into the candidate dash.

**Pull request merged**